retux/client/resources/auto_moderation.py

```diff
@@ -0,0 +1,223 @@
+    An enumeration of the different trigger types of Discord's auto moderation
+
+    ---
+    The `HARMFUL_LINK` and `SPAM` trigger types are not yet released, so they cannot be used in most servers.
+    ---
+
+
+    Constants
+    ---------
+    KEYWORD
+        Check if content contains words from a user defined list of keywords.
+    HARMFUL_LINK
+        Check if content contains any harmful links.
+    SPAM
+        Check if content represents generic spam.
+    KEYWORD_PRESET
+        Check if content contains words from internal pre-defined wordsets.
+    """
+
+    KEYWORD = 1
+    """Check if content contains words from a user defined list of keywords."""
+    HARMFUL_LINK = 2
+    """Check if content contains any harmful links."""
+    SPAM = 3
+    """Check if content represents generic spam."""
+    KEYWORD_PRESET = 4
+    """Check if content contains words from internal pre-defined wordsets."""
+
+
+class AutoModerationActionType(IntEnum):
+    """
+    An enumeration representing the different types for an auto-moderation action.
+
+    Constants
+    ---------
+    BLOCK_MESSAGE
+        Blocks the content of a message according to the rule.
+    SEND_ALERT_MESSAGE
+        Logs user content to a specified channel.
+    TIMEOUT
+        Timeout user for a specified duration.
+
+        Can only be setup for KEYWORD rules. The MODERATE_MEMBERS permission is required to use this action type.
+    """
+
+    BLOCK_MESSAGE = 1
+    """Blocks the content of a message according to the rule."""
+    SEND_ALERT_MESSAGE = 2
+    """Logs user content to a specified channel."""
+    TIMEOUT = 3
+    """Timeout user for a specified duration."""
+
+
+@define()
+class AutoModerationActionMetadata:
+    """
+    A class object representing additional metadata of an auto-moderation action.
+
+    Attributes
+    ----------
+    channel_id : `Snowflake`
+        The ID of the Channel to which user content should be logged.
+    duration_seconds : `int`
+        Timeout duration in seconds.
+
+        Maximum is 2419200 seconds (4 weeks)
+    """
+
+    channel_id: str | Snowflake = field(converter=Snowflake)
+    """The ID of the Channel to which user content should be logged."""
+    duration_seconds: int = field()
+    """Timeout duration in seconds."""
+
+
+@define()
+class AutoModerationAction:
+    """
+    A class object representing an auto-moderation action what will be executed when an auto-moderation rule triggers.
+
+    Attributes
+    ----------
+    type : `AutoModerationActionType`
+        The type of action.
+    metadata : `AutoModerationActionMetadata`, optional
+        Additional metadata needed during execution for this specific action type, if any.
+    """
+
+    type: int | AutoModerationActionType = field(converter=AutoModerationActionType)
+    """The type of action."""
+    metadata: int | AutoModerationActionMetadata | None = field(
+        converter=optional_c(list_c(AutoModerationActionMetadata)), default=None
```

Comment:
```suggestion
        converter=optional_c(AutoModerationActionMetadata), default=None
```

Developer documentation suggests that there is no list of metadata.